### PR TITLE
Add show interest form

### PR DESCRIPTION
**Pull request by dimitrismistriotis**

### CODE REVIEW

File: dj_register_interest/tests/test_register_interest_view.py [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+        response = client.post(self._TARGET_URL, form_data)
+
+        # Assuming you redirect to a success page on valid data
+        assert response.status_code == 302
+
+    def test_interest_view_post_invalid_data(self, client: Client) -> None:
+        form_data = {}  # Invalid data
+        response = client.post(self._TARGET_URL, form_data)
+
+        assert response.status_code == 200  # Stays on the same page due to form errors
```
Comment (menkotoglou) [inline dj_register_interest/tests/test_register_interest_view.py:37]: This looks a bit odd. A `status_code == 200` for an unsuccessful event.

I would also suggest that we use `pytest.mark.parametrize` to test for different errors in form_data.

File: dj_register_interest/views.py [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+        """Get request for the view."""
+        form = self.form_class()
+        return render(request, self.template_name, {"form": form})
+
+    def post(self, request, *args, **kwargs) -> HttpResponseRedirect | HttpResponse:
+        """Post request for the view."""
+        form = self.form_class(request.POST)
+        if form.is_valid():
+            form.save()
+            messages.success(request, "Thank you for registering your interest!")
```
Comment (menkotoglou) [inline dj_register_interest/views.py]: May nit picking, but I think that this should be moved into a separate function, where a try/catch block exists and saves the data to the db.

I suggest endpoints handle the networking part, while usecases/handlers or we can call them as we want, are the ones which should be handling db logic.

**Review by menkotoglou — COMMENTED** [review]
Great work overall, added a couple of comments.

File: dj_register_interest/tests/test_register_interest_view.py (hunk 1 above)
Comment (dimitrismistriotis) [reply dj_register_interest/tests/test_register_interest_view.py:37]: `# Stays on the same page due to form errors` This is not an API response. BTW forgot to mention that testing code is modified from synthetic.

File: dj_register_interest/views.py (hunk 2 above)
Comment (dimitrismistriotis) [reply dj_register_interest/views.py]: Hmmm. Interesting point which opens many paths on how to take it from here.

My general approach **was** that for trivial things it is OK to have them embedded in the view specially here that we have a class based view. There is not for example a notification message being dispatched to the owners of the app.

Could be that we have a "policy" - then we should put it in `Contributing.md` - that there should always be a "middleware" code (as in middle between web and database or other endpoints) to handle such workflows even if its actual functionality is minimal.

... in which case I'd say we can create an issue and discuss it there so that we can have a reference point.

**Comment by dimitrismistriotis** [comment]
Last commit solves: https://github.com/dimitrismistriotis/djforge/issues/22

File: dj_register_interest/handlers.py [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+    Can be expanded to notify project's stakeholders with email, text message, etc.
+    """
+
+    def __call__(self, form: InterestForm) -> bool:
+        """Entry point"""
+        if form.is_valid():
+            form.save()
+            return True
+
+        return False
```
Comment (menkotoglou) [inline dj_register_interest/handlers.py]: A beauty :cupid:

**Review by menkotoglou — APPROVED** [review]
Great one!

File: dj_register_interest/views.py (hunk 2 above)
Comment (menkotoglou) [reply dj_register_interest/views.py]: Indeed, I'd say it sounds sth like we should be adding to our `Contributing.md` for reference.

File: dj_register_interest/tests/test_register_interest_view.py (hunk 1 above)
Comment (dimitrismistriotis) [reply dj_register_interest/tests/test_register_interest_view.py:37]: Asked ChatGPT :)

```
Should a Django request, should a Django POST request return 200 even if there was a mistake or something erroneous while unprocessing the form?
ChatGPT
In Django, the HTTP status code returned for a POST request, or any request, should accurately reflect the result of that request. Here's a guideline:

HTTP 200 OK: This status code indicates that the request has succeeded. It's typically used when the server has successfully processed the request and is sending a response back to the client. If a Django POST request processes the form successfully, even if the form data is not valid (e.g., missing fields, invalid data), but the server processed the request without any server-side errors, returning a 200 status code is appropriate. This often happens when the form is returned to the user with validation errors highlighted.

HTTP 4XX Codes (Client Errors): If there's a mistake or error that's attributable to the client's request (like a bad request format, invalid data, etc.), a 4XX series error is more appropriate. For example:

HTTP 400 Bad Request: This status code is used when the request sent by the client was invalid or malformed.
HTTP 422 Unprocessable Entity (if you're following WebDAV extensions): Suitable for scenarios where the server understands the content type of the request entity, and the syntax of the request entity is correct, but it was unable to process the contained instructions.
HTTP 5XX Codes (Server Errors): If an error occurs on the server while processing the request, which is not due to the client's request, a 5XX status code should be used. For example, HTTP 500 Internal Server Error is used when an unexpected condition was encountered on the server and no more specific message is suitable.

In summary, if the form processing encounters an error due to client-side issues (like validation errors), a 200 OK response can be returned along with the form displaying the errors. However, if the error is due to a server-side issue, a 5XX error would be more appropriate. The key is to choose a status code that accurately represents the outcome of the request.
```

File: dj_register_interest/views.py (hunk 2 above)
Comment (dimitrismistriotis) [reply dj_register_interest/views.py]: Since PR got accepted, decided to capture this in an issue, https://github.com/dimitrismistriotis/djforge/issues/24 , and do it separately
